Angular-architects/ddd:ui customization to fulfill AdUp requirements.

waismeeran opened the pull request:
Ticket: https://adup-tech.atlassian.net/browse/AUS-489

summary:

1. Resolved component generation discrepancies with consistent naming conventions.
2. Integrated linting tools (ESLint, Stylelint, Addlint) into the plugin.
3. Implemented README enhancement for standardized documentation structure.
4. Applied default host class with display block to SCSS files.
5. Customized Angular library (hostclass, viewEncapsulation and changeDetection) as specified.
6. Integrated Storybook for UI component development.
7. Automated harness tests for UI components functionality.
8. Default --additionalPrefix set to as- to avoid conflicts with other plugins.
9. Implemented public-api and exported it in index.ts for improved module accessibility.

![image](https://github.com/adup-tech/nx-ddd-plugin/assets/5171452/aed1c37d-cd29-4a6d-9a20-8e6c2a76c006)

![image](https://github.com/adup-tech/nx-ddd-plugin/assets/5171452/87871a75-8855-4551-ae0f-1a785cec1736)
![image](https://github.com/adup-tech/nx-ddd-plugin/assets/5171452/40c3d6c0-aa64-48be-be45-09924292fbbf)

Superfantastisch reviewed (commented):
It looks good to me. But I can't really say wether it does what it is supposed to do :D

I just want to mention that I don't like some of the coding styles here. Especially that values changes/ object manipulation are encapsulated within functions like:
```typescript
/**
 * Reset the index file
 */
export function resetIndexFile(tree: Tree, indexFilePath: string) {
    const contents = tree.read(indexFilePath, 'utf-8');
    const rest = contents.split('\n').slice(1);
    tree.write(indexFilePath, (rest || []).join('\n'));
}
```
I think util functions always should return a value and be tested.
It is not visible from outside of this function that tree is getting manipulated here.
This could lead to side effects and errors (when an other developer comes in the main file and makes changes). It also makes the code hard to read, hard to debug (you always have to step in) and hard to test.
But since this pattern is inside this project from the origin I think it is okay.

I would prefer something like:
```typescript
/**
 * Reset the index file
 */
export function resetIndexFile(tree: Tree, indexFilePath: string): string {
    const contents = tree.read(indexFilePath, 'utf-8');
    const rest = contents.split('\n').slice(1);
    return (rest || []).join('\n'));
}
```
Or
```typescript
/**
 * Reset the index file
 */
export function resetIndexFile(tree: Tree, indexFilePath: string): Tree {
    const contents = tree.read(indexFilePath, 'utf-8');
    const rest = contents.split('\n').slice(1);
    tree.write(...)
    return tree;
}
```
It would be nice if you could add return values to the typescript functions. Void when nothing is returned like in the example above.

In the original project there are no tests at all. But since you created a lot of new util functions you maybe could generate some test for them. At least for the (pure) functions which are returning values.

waismeeran commented:
> It looks good to me. But I can't really say wether it does what it is supposed to do :D
> 
> I just want to mention that I don't like some of the coding styles here. Especially that values changes/ object manipulation are encapsulated within functions like:
> 
> ```ts
> /**
>  * Reset the index file
>  */
> export function resetIndexFile(tree: Tree, indexFilePath: string) {
>     const contents = tree.read(indexFilePath, 'utf-8');
>     const rest = contents.split('\n').slice(1);
>     tree.write(indexFilePath, (rest || []).join('\n'));
> }
> ```
> 
> I think util functions always should return a value and be tested. It is not visible from outside of this function that tree is getting manipulated here. This could lead to side effects and errors (when an other developer comes in the main file and makes changes). It also makes the code hard to read, hard to debug (you always have to step in) and hard to test. But since this pattern is inside this project from the origin I think it is okay.
> 
> I would prefer something like:
> 
> ```ts
> /**
>  * Reset the index file
>  */
> export function resetIndexFile(tree: Tree, indexFilePath: string): string {
>     const contents = tree.read(indexFilePath, 'utf-8');
>     const rest = contents.split('\n').slice(1);
>     return (rest || []).join('\n'));
> }
> ```
> 
> Or
> 
> ```ts
> /**
>  * Reset the index file
>  */
> export function resetIndexFile(tree: Tree, indexFilePath: string): Tree {
>     const contents = tree.read(indexFilePath, 'utf-8');
>     const rest = contents.split('\n').slice(1);
>     tree.write(...)
>     return tree;
> }
> ```
> 
> It would be nice if you could add return values to the typescript functions. Void when nothing is returned like in the example above.
> 
> In the original project there are no tests at all. But since you created a lot of new util functions you maybe could generate some test for them. At least for the (pure) functions which are returning values.

Many thanks for reviewing my code. Indeed it's a good practice for utility functions to consistently return a value whenever it makes sense within the context of their purpose. However, there might be scenarios where a utility function performs a side effect without needing to return a specific value. For instance, a function that logs information or modifies a global state might not need to return a value explicitly.

In my case, 
`updateFileContent`: This function writes updated content to a file. It doesn't necessarily need to return a value since its primary task is performing the file write operation. It operates by side effect, altering the file directly.

`resetIndexFile:`  Similarly to `updateFileContent`, this function doesn't explicitly need to return a value. Its purpose is to reset the index file by modifying it directly.

Regarding the test, yes, I thought about this, but since the original plugin does not have any test, I decided to save time and ignore the test. However, if you think it is necessary to have a test, then I have no problem creating one.
